src/main/java/com/artipie/api/RepositoryNameValidator.java

```diff
@@ -34,16 +34,11 @@ public ValidRepositoryName(final String name) {
      * Ctor.
      * @param name The name to test
      */
-    public ValidRepositoryName(final RepositoryName name) {
+    public RepositoryNameValidator(final RepositoryName name) {
         this(name.toString());
     }
 
-    /**
-     * Checks if the repository name is valid. The name is considered valid if it does
-     * not contain reserved words `_storages, _permissions, _credentials` in it.
```

Comment:
I'd suggest to move the explanation about valid repository name into class javadoc, thus this class' user can just read the doc to understand what name is considered valid instead of reading the whole class code.

Reply:
moved